frontend/src/components/CardDetails/CardDetails.test.tsx

```diff
@@ -1,58 +1,116 @@
 
 import { CardDetails } from "./CardDetails";
 
-const mockProps = {
-  productTitle: "Pams Butter",
-  price: "3.50",
-  promo: null,
-  store: "Woolworths",
-};
-
 describe("Given the user is looking at an individual products's details", () => {
   describe("When the product is displayed on the home page", () => {
     it("Then the store, product name and price is displayed", () => {
-      renderWithRouter(<CardDetails {...mockProps} />);
+      renderWithRouter(<CardDetails product={nwProduct} />);
 
-      expect(screen.getByText(/pams butter/i)).toBeInTheDocument();
-      expect(screen.getByText(/woolworths/i)).toBeInTheDocument();
-      expect(screen.getByText("$3.50")).toBeInTheDocument();
+      expect(
+        screen.getByText(
+          /Chia Sisters Gut Lemon & Golden Kiwifruit Sparkling Drink 250ml/i,
+        ),
+      ).toBeInTheDocument();
+      // TODO expect(screen.getByText(/new world/i)).toBeInTheDocument(); When the shop code is fixed
+      expect(screen.getByText("$5.59")).toBeInTheDocument();
     });
   });
 
   describe("When the product is displayed on the browse page", () => {
     it("Then product name and price is displayed", () => {
-      renderWithRouter(<CardDetails {...mockProps} />, {
+      renderWithRouter(<CardDetails product={nwProduct} />, {
         route: "/browse",
         path: "/browse",
       });
 
-      expect(screen.queryByText(/woolworths/i)).not.toBeInTheDocument();
-      expect(screen.getByText(/pams butter/i)).toBeInTheDocument();
-      expect(screen.getByText("$3.50")).toBeInTheDocument();
+      expect(screen.queryByText(/new world/i)).not.toBeInTheDocument();
+      expect(
+        screen.getByText(
+          /Chia Sisters Gut Lemon & Golden Kiwifruit Sparkling Drink 250ml/i,
+        ),
+      ).toBeInTheDocument();
+      expect(screen.getByText("$5.59")).toBeInTheDocument();
     });
   });
 
   describe("When the button is rendered", () => {
     it("Then the text add to cart is displayed", async () => {
-      renderWithRouter(<CardDetails {...mockProps} />);
+      renderWithRouter(<CardDetails product={nwProduct} />);
 
-      const button = screen.getByRole("button");
-      expect(button).toHaveTextContent(/add to cart/i);
+      const button = screen.getByRole("button", { name: /add to cart/i });
+      expect(button).toBeInTheDocument();
     });
-
-    it.todo("Then it adds item to cart");
   });
 
   describe("When the product details are displayed", () => {
     it("Then it has no accessibility violations", async () => {
-      const { container } = renderWithRouter(<CardDetails {...mockProps} />);
+      const { container } = renderWithRouter(
+        <CardDetails product={nwProduct} />,
+      );
 
       const results = await axe(container);
       expect(results).toHaveNoViolations();
     });
   });
+
+  // --- INTERACTIONS --- //
+
+  describe("when the add to cart button is displayed", () => {
```

Comment:
```suggestion
  describe("When the add to cart button is displayed", () => {
```

❓ You have two tests described as: 
"When the add to cart button is displayed" and just above "When the button is rendered"
What is the difference? Should these be under the same 'When' heading or a clearer separation?

Reply:
I tried to seperate them more specifically now, so that adding when clicked is nested under when the button is rendered. And the undo button is under a different describe function